tests/chart_tests/test_byo_sa.py

```diff
@@ -0,0 +1,31 @@
+import pytest
+
+from tests import supported_k8s_versions
+from tests.chart_tests.helm_template_generator import render_chart
+
+
+@pytest.mark.parametrize(
+    "kube_version",
+    supported_k8s_versions,
+)
+class TestServiceAccounts:
+    def test_serviceaccount_rbac_disabled(self, kube_version):
+        docs = render_chart(
+            kube_version=kube_version, values={"global": {"rbacEnabled": False}, "nats": {"nats": {"createJetStreamJob": False}}}
+        )
+
+        # Check that the Deployment or StatefulSet is using the default ServiceAccount
+        sa_name = ""
+        for doc in docs:
+            if doc.get("kind") == "ServiceAccount":
+                assert (
+                    sa_name == doc["metadata"]["name"]
+                ), f"No ServiceAccounts were expected when rbacEnabled is False, but got {sa_name}"
```

Comment:
The assertion in this block is misleading. It's asserting about the contents of `doc["metadata"]["name"], but that is not the problem. The problem is that there is any service account at all.

IMHO the better way to handle this would be to check that there are no ServiceAccount entities, and report about which ones are found.

```py
service_account_names = [doc['metadata']['name'] for doc in docs if doc['kind'] == 'ServiceAccount']
assert not doc, f'Expected no ServiceAccounts but found {service_account_names}'
```

Reply:
makes sense